### PR TITLE
rename `Setup` to `Configure`

### PR DESCRIPTION
**Pull request by aboedo**
Renamed `Setup` to `Configure`. 

I added a deprecation to `Setup` although it was never technically public in a non-prerelease build, because it doesn't hurt to have it and it'll make it slightly easier for anyone who was using a pre-release build to update.

### CODE REVIEW

File: RevenueCat/Scripts/Purchases.cs [hunk 1]
```diff
@@ -41,56 +41,76 @@ public partial class Purchases : MonoBehaviour
     /// </summary>
     public delegate void GetPromotionalOfferFunc(PromotionalOffer promotionalOffer, Error error);
 
-    [Tooltip("Activate if you plan to call Purchases.Setup programmatically.")]
+    [Tooltip("Activate if you plan to call Purchases.Configure or Purchases.Setup programmatically.")]
```
Comment (aboedo) [inline RevenueCat/Scripts/Purchases.cs:44]: did a decent amount of cleanup on these tooltips so they look a bit better

Comment (aboedo) [reply RevenueCat/Scripts/Purchases.cs:44]: mostly line breaks both in C# by splitting the string and in the tooltip itself with `\n`

File: RevenueCat/Scripts/Purchases.cs [hunk 2]
```diff
@@ -134,10 +154,17 @@ private void Setup(string newUserId)
             .SetUseAmazon(useAmazon)
             .SetDangerousSettings(dangerousSettings);
 
-        Setup(builder.Build());
+        Configure(builder.Build());
     }
 
+    [Obsolete("Deprecated, use Configure instead.", false)]
```
Comment (aboedo) [inline RevenueCat/Scripts/Purchases.cs:160]: this method was never public in a final release, but it doesn't hurt to have the deprecation for anyone using a release candidate

File: RevenueCat/Scripts/PurchasesConfiguration.cs [hunk 3]
```diff
@@ -41,6 +52,7 @@ public static Builder Init(string apiKey)
 
             public PurchasesConfiguration Build()
             {
+                _dangerousSettings ??= new DangerousSettings(false);
```
Comment (aboedo) [inline RevenueCat/Scripts/PurchasesConfiguration.cs:55]: without this, creating a build but not setting dangerous settings would crash

Comment (tonidero) [reply RevenueCat/Scripts/PurchasesConfiguration.cs:55]: I guess we don't have tests for this right?

File: RevenueCat/Scripts/Purchases.cs (hunk 2 above)
Comment (tonidero) [reply RevenueCat/Scripts/Purchases.cs:160]: Well... I guess the only downside is that we will have an extra deprecated function to test/remove at some point... But then again, I think what you said makes sense so I'm ok with this 👍

**Review by tonidero — APPROVED** [review]
Newbie here but LGTM!

**Review by vegaro — APPROVED** [review]

File: RevenueCat/Scripts/PurchasesConfiguration.cs (hunk 3 above)
Comment (aboedo) [reply RevenueCat/Scripts/PurchasesConfiguration.cs:55]: we don't have unit tests for anything in Unity, sadly. Hoping to get that started soon. We do have API tests, as well as integration tests that export a `.unityPackage`, import it into an iOS and an Android app, and try building (and creating an archive on iOS)

**Pull request merged**